lib/LaTeXML/Core/Gullet.pm

```diff
@@ -101,17 +101,30 @@ sub flush {
   $$self{mouthstack} = [];
   return; }
 
-# Do something, while reading stuff from a specific Mouth.
+# Do something, while reading stuff from a specific source Mouth.
+# If $source is a string or Tokens, a Mouth will be created to process it.
 # This reads ONLY from that mouth (or any mouth openned by code in that source),
 # and the mouth should end up empty afterwards, and only be closed here.
 sub readingFromMouth {
-  my ($self, $mouth, $closure) = @_;
+  my ($self, $source, $closure) = @_;
+  my $mouth;
+  my $intokens;
+  my $sourcetype = ref $source;
+  if (! $sourcetype) {          # for a string, create Mouth to read its content
+    $mouth = LaTeXML::Core::Mouth->new($source); }
+  elsif ($sourcetype =~ /^LaTeXML::Core::Tokens?$/) { # Tokens will be unread into Mouth
+    $mouth = LaTeXML::Core::Mouth->new();
+    $intokens = $source; }
+  else {
+    $mouth = $source; }         # Hopefully is a Mouth.
```

Comment:
Could we check that it is indeed a Mouth and emit an Error or Fatal otherwise?

Reply:
You want more than a "Hopefully" comment?  Sure, next commit fixes that :>